src/modules-by-example.adoc

```diff
@@ -319,3 +320,472 @@ Here again, Ion 1.1 enables a new technique: we can prepend new symbols to the e
+      (*macro* scatterplot [(point\... points)]
+        [points]))
+  ],
+  install_macros: [ chart ]                        // <2>
+}
+----
+
+<1> Importing `geo` makes its macros accessible within `chart`.
+<2> The `geo` module is not installed into the encoding environment, so its macros are not
+accessible in the document body.
+
+
+=== Macro Aliases
+
+We've seen how to resolve an ambiguous macro name by using qualified references.  Another approach
+is to give new names to existing macros.  Suppose we want to add a 3d chart to our module, so we
+import both the 2d and 3d modules:
+
+[{nrm}]
+----
+*$ion_1_1*
+**$ion_encoding**::{
+  modules: [
+    (*module* chart
+      (*import* geo "com.example.geometry" 1)
+      (*import* g3d "com.example.graphics.3d" 1)
+      (*macro* scatterplot [(point\... points)]
+
+  ⇒ **error**: 'point' is ambiguous, exported by 'geo' and 'g3d'.
+----
+
+The most direct way to fix this is to use a qualified reference.  We've seen this used in
+E-expressions like `(:geo:point 17 28)`, but now we need it in a signature where the special
+smile syntax does not apply.  Instead, use a quoted symbol:
+
+[{nrm}]
+----
+      (*macro* scatterplot [(':geo:point' \... points)]
+        [points]))
+----
+
+That has the intended effect of keeping `scatterplot` using 2D points, but it's somewhat awkward.
+A more ergonomic approach is to introduce an alias to disambugate:
+
+
+[{nrm}]
+----
+  modules: [
+    (*module* chart
+      (*import* geo "com.example.geometry" 1)
+      (*import* g3d "com.example.graphics.3d" 1)
+      (*alias* point2 ':geo:point')               // <1>
+      (*macro* scatterplot [(point2 \... points)]  // <2>
+        [points])
+      ...
+----
+
+<1> Declaration of alias `point2`.
+<2> Use of that new name in a signature.
+
+Aliases can only be declared within a module, where they can be used wherever a macro reference
+occurs, including for macro invocations in the template language.  In addition to disambiguation,
+they can be used to shorten long names, or to give names to anonymous macros.
+
+
+=== Exports
+
+Unlike `macro` definitions, aliases are not automatically exported from the module where they are
+declared; they are presumed to be implementation details.  Sometimes it's helpful to make them
+available to consumers of the module, and for that they can be exported:
+
+[{nrm}]
+----
+*$ion_1_1*
+**$ion_encoding**::{
+  modules: [
+    (*import* geo "com.example.geometry" 1),
+    (*import* g3d "com.example.graphics.3d" 1),
+    (*module* local
+      (*import* geo)
+      (*import* g3d)
+      (*alias* point2 ':geo:point')
+      (*alias* point3 ':g3d:point')
+      (*export* point2 point3))
+  ],
+  install_macros: [ local, geo, g3d ]
+}
+(:point2 93 5)
+(:point3 0 12 33)
+----
+
+Exports can also be used to "pass through" selected macros from an imported module: `(*export*
+':g2d:line')` exports the name `line` from the enclosing module.
```

Comment:
I think this is different than:
```
(alias line ':g2d:line')
(export line)
```

And I think the difference is I don't get a local name, but is that right (and this question makes me think we should have text that illustrates if this is a difference or not).

Reply:
You are correct. It has the same effect to the module's consumer, but there's no local name.  I'll add the example.